spec.html

```diff
@@ -1085,10 +1085,11 @@ <h1>Value Notation</h1>
+      <p>From the perspective of this specification, the word “is” is used to compare two values for equality, as in “If _bool_ is *true*, then ...”, and the word “contains” is used to search for a value inside lists using equality comparisons, as in "If _list_ contains a Record _r_ such that _r_.[[Foo]] is *true*, then ...". The <em>specification identity</em> of values determines the result of these comparisons and is axiomatic in this specification.</p>
+      <p>From the perspective of the ECMAScript language, language values are compared for equality using the SameValue abstract operation and the abstract operations it transitively calls. The algorithms of these comparison abstract operations determine <em>language identity</em> of ECMAScript language values.</p>
+      <p>For specification values, examples of values without specification identity include, but are not limited to: mathematical values and extended mathematical values; ECMAScript source text, surrogate pairs, Directive Prologues, etc; UTF-16 code units; Unicode code points; enums; abstract operations, including syntax-directed operations, host hooks, etc; and ordered pairs. Examples of specification values with specification identity include, but are not limited to: any kind of Records, including Property Descriptors, PrivateElements, etc; Parse Nodes; Lists; <emu-xref href="#sec-set-and-relation-specification-type">Sets</emu-xref> and Relations; Abstract Closures; Data Blocks; Private Names; execution contexts and execution context stacks; agent signifiers; and WaiterLists. These lists are not exhaustive.</p>
+      <p>The ECMAScript language values without specification identity and without language identity are <emu-xref href="#sec-ecmascript-language-types-undefined-type">*undefined*</emu-xref>, <emu-xref href="#sec-ecmascript-language-types-null-type">*null*</emu-xref>, <emu-xref href="#sec-ecmascript-language-types-boolean-type">Booleans</emu-xref>, <emu-xref href="#sec-ecmascript-language-types-string-type">Strings</emu-xref>, <emu-xref href="#sec-ecmascript-language-types-number-type">Numbers</emu-xref>, and <emu-xref href="#sec-ecmascript-language-types-bigint-type">BigInts</emu-xref>. The ECMAScript language values with specification identity and language identity are the <emu-xref href="#sec-object-type">Objects</emu-xref>. <emu-xref href="#sec-ecmascript-language-types-symbol-type">Symbols</emu-xref> are a special case. All Symbol values have specification identity. Symbol values produced by <emu-xref href="#sec-symbol.for">Symbol.for</emu-xref> do not have language identity, while Symbol values not produced by <emu-xref href="#sec-symbol.for">Symbol.for</emu-xref> have language identity.</p>
```

Comment:
I am OK with this, but I think is a little awkward to read. I think I'd do it a little differently:

> Specification identity agrees with language identity for all ECMAScript language values except Symbol values produced by `Symbol.for`. The ECMAScript language values without specification or language identity are (list). The ECMAScript language values with specification and language identity are (list). Symbol values produced by `Symbol.for` have language identity, but not specification identity.

I guess the main thing I want is a sentence up front saying that they agree in all cases except Symbol.for symbols; the other wording tweaks are less important.

Reply:
WFM